### ISSUE TITLE
Usage with storybook

### ISSUE THREAD

**Issue by archansel** [scottrippey/next-router-mock#10]
Hi, you mention this library can be used with storybook but I don't see any documentation for it, is it using MemoryRouter export and put it in component tree?

**Comment by scottrippey** [comment]
I didn't realize that the docs still said Storybook!  Sorry for the confusion.
As of now, I have NOT tested this repo with Storybook at all, and I'm not totally sure how it can be done.

Probably the easiest way to use it in Storybook is to provide a custom webpack config to alias `next/router`  to `next-router-mock` ... which is very similar to the docs for using `jest.mock('next/router', () => require('next-router-mock'))`.  
Let me think about it a bit more, and I'll determine if this is truly the best approach.  If so, I can document it.

Another approach to consider is to have a "provider-based" approach for Storybook, where the router could be supplied via a context ... that way we could better mock the current router state.  I'll see if this is possible too.

**Comment by scottrippey** [comment]
So, this might be a temporary solution ... I verified it works with Next v10 (but I think v11 might have a different path for importing the context):
```
import { RouterContext } from "next/dist/next-server/lib/router-context";
import { useRouter } from 'next-router-mock';

export const MemoryRouter = ({ children }) => {
  return <RouterContext.Provider value={useRouter()}>{children}</RouterContext.Provider>;
}
```

Then, you can either wrap individual stories with `<MemoryRouter>` or you can globally wrap all stories by adding this as a decorator to `storybook/preview.js`.
Examples:

Wrap individual stories:
```
// LinkButton.stories.jsx
export const LinkButtonExample = () => (
  <MemoryRouter>
    <LinkButton href="/foo">Go to /foo</LinkButton>
  </MemoryRouter>
);
```
or to wrap ALL Storybook stories:
```
// storybook/preview.js
import { addDecorator } from '@storybook/react';

addDecorator(Story => <MemoryRouter><Story /></MemoryRouter>)
```


This approach will support both `next/link`, as well as any usages of `useRouter()`.

I'm currently thinking about how to improve all of this, too.

**Comment by scottrippey** [comment]
I've created a `<MemoryRouterProvider>` which can be used in Storybook.  Take a look at the PR here! https://github.com/scottrippey/next-router-mock/pull/15

**Comment by scottrippey** [comment]
Fixed in #15  , published in `v0.4.0`
